perfdash/github-configs-fetcher.go

```diff
@@ -0,0 +1,77 @@
+/*
+Copyright 2015 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package main
+
+import (
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"strings"
+
+	"gopkg.in/yaml.v2"
+)
+
+type githubDirContent struct {
+	Name        string `yaml:"name"`
+	Path        string `yaml:"path"`
+	DownloadUrl string `yaml:"download_url"`
+	Type        string `yaml:"type"`
+	Url         string `yaml:"url"`
+}
+
+/*
+Gets config paths from github directory. It uses github API, which is documented
+here: https://developer.github.com/v3/repos/contents/
+
+Example url: https://api.github.com/repos/pierewoj/test-infra/contents/config/jobs/kubernetes/sig-release/release-branch-jobs
+
+Different branch can be specified by appending "?ref=branch-name" parameter at the end
+of the url.
+*/
+func GetConfigsFromGithub(url string) ([]string, error) {
+	var result []string
+	contents, err := getGithubDirContents(url)
+	if err != nil {
+		return nil, err
+	}
+	for _, c := range contents {
+		// Dirs and non-yaml files are ignored; this means that there is no
+		// recursive search, it should be good enough for now.
+		if c.Type == "file" && strings.HasSuffix(c.Name, ".yaml") {
+			result = append(result, c.DownloadUrl)
+		}
+	}
+	return result, nil
+}
+
+func getGithubDirContents(url string) ([]githubDirContent, error) {
+	fmt.Printf("Downloading github spec from %v\n", url)
```

Comment:
Use klog/log?

Here and other places

Reply:
That's a good idea but we use fmt.Printf in other places. If we wan't to use klog, I'd migrate all logging places at once (if we don't do that, some lines will have timestamp, some won't etc).

Let's do this in a separate PR.

Reply:
https://github.com/kubernetes/perf-tests/issues/929